AndroidTest/app/src/main/java/cloudant/com/androidtest/MyActivity.java

```diff
@@ -48,78 +49,115 @@
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
+        //Stop the tests being re-run when we are returned to this activity
+        if(savedInstanceState == null) {
 
-        setTestProperties();
+            super.onCreate(savedInstanceState);
```

Comment:
`super.onCreate(savedInstanceState);` should be called before we do our stuff, we always have to call the super to prevent exceptions being thrown see: http://developer.android.com/reference/android/app/Activity.html#onCreate%28android.os.Bundle%29